FWCore/Framework/interface/ComponentFactory.h

```diff
@@ -34,100 +34,94 @@
+
+          if (wm.get() == nullptr) {
+            Exception::throwThis(errors::Configuration, "UnknownModule", T::name().c_str(), " of type ",
+                                 modtype.c_str(),
+                                 " has not been registered.\n"
+                                 "Perhaps your module type is misspelled or is not a "
+                                 "framework plugin.\n"
+                                 "Try running EdmPluginDump to obtain a list of "
+                                 "available Plugins.");
+          }
+
+          //cerr << "Factory: created the worker" << endl;
+
+          std::pair<typename MakerMap::iterator, bool> ret =
+              makers_.insert(std::pair<std::string, std::shared_ptr<Maker> >(modtype, wm));
+
+          if (ret.second == false) {
+            Exception::throwThis(errors::Configuration, "Maker Factory map insert failed");
+          }
+
+          it = ret.first;
+        }
+
+        try {
+          return convertException::wrap([&]() -> std::shared_ptr<base_type> {
+            return it->second->addTo(esController, iProvider, iConfiguration, replaceExisting);
+          });
+        } catch (cms::Exception& iException) {
+          std::string edmtype = iConfiguration.template getParameter<std::string>("@module_edm_type");
+          std::string label = iConfiguration.template getParameter<std::string>("@module_label");
+          std::ostringstream ost;
+          ost << "Constructing " << edmtype << ": class=" << modtype << " label='" << label << "'";
+          iException.addContext(ost.str());
+          throw;
+        }
+        return std::shared_ptr<base_type>();
       }
-   
+
       // ---------- static member functions --------------------
       static ComponentFactory<T> const* get();
 
       // ---------- member functions ---------------------------
 
-   private:
-      
-      ComponentFactory(const ComponentFactory&); // stop default
+    private:
+      ComponentFactory(const ComponentFactory&);  // stop default
 
-      const ComponentFactory& operator=(const ComponentFactory&); // stop default
+      const ComponentFactory& operator=(const ComponentFactory&);  // stop default
 
       // ---------- member data --------------------------------
       mutable MakerMap makers_;
-};
-
-   }
-}
-#define COMPONENTFACTORY_GET(_type_) \
-EDM_REGISTER_PLUGINFACTORY(edmplugin::PluginFactory<edm::eventsetup::ComponentMakerBase<_type_>* ()>,_type_::name()); \
-static edm::eventsetup::ComponentFactory<_type_> const s_dummyfactory; \
-namespace edm { namespace eventsetup { \
-template<> edm::eventsetup::ComponentFactory<_type_> const* edm::eventsetup::ComponentFactory<_type_>::get() \
-{ return &s_dummyfactory; } \
-  } } \
-typedef int componentfactory_get_needs_semicolon
+    };
+
+  }  // namespace eventsetup
+}  // namespace edm
+#define COMPONENTFACTORY_GET(_type_)                                                                                  \
```

Comment:
Does this work? It removed the line continuation!

Reply:
Isn't it just moved all to way the right ?

Reply:
I retract my statement :). Thanks Andrea!